src/main/java/org/usfirst/frc/team4099/robot/subsystems/Arm.kt

```diff
@@ -0,0 +1,163 @@
+        slaveSRX3.set(ControlMode.Follower, Constants.Arm.MASTER_SRX_ID.toDouble())
+        if (useVelocityControl) {
+            masterSRX.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 10)
+            masterSRX.setSensorPhase(false)
+            masterSRX.configNominalOutputForward(+0.0, 0)
+            masterSRX.configPeakOutputForward(+1.0, 0)
+            masterSRX.selectProfileSlot(0, 0)
+            masterSRX.config_kF(0, 0.0, 0)
+            masterSRX.config_kP(0, 0.0, 0)
+            masterSRX.config_kI(0, 0.0, 0)
+            masterSRX.config_kD(0, 0.0, 0)
+        }
+    }
+
+    /**
+     * Sets the control mode to Velocity Control or Movement Control
+     *
+     * @param velocityControl a Boolean for setting to Velocity Control
+     */
+    fun velocityControlSwitch(velocityControl: Boolean) {
+        useVelocityControl = velocityControl
+    }
+
+    /**
+<<<<<<< HEAD
+     * Changes power to the arm
+     * @param[power] double power
+=======
+     * Changes the power to the Arm
+     *
+     * @param power A double for the power going to the arm
+>>>>>>> a38ca8e94cc356d6aad10ea10931c1e2a57a6cf7
+     */
+    private fun setArmPower(power: Double) {
+        masterSRX.set(ControlMode.MotionMagic, Math.abs(power))
+    }
+
+    /**
+     * Outputs arm subsystem
+     *
+     */
+    override fun outputToSmartDashboard() {
+        SmartDashboard.putNumber("armPower", armPower)
+    }
+
+    @Synchronized override fun stop() {
+        movementState = MovementState.STATIONARY
+        setArmPower(0.0)
+    }
+
+    /**
+     * Handles power to arm during each loop cycle
+     *
+     * @constructor Creates loop that controls power to arm during each loop cycle
+     */
+    val loop: Loop = object : Loop {
+        override fun onStart() {
+            setArmPower(0.0)
+        }
+
+        /**
+         * Handles events during each loop cycle, depends on arm control mode
+         */
+        override fun onLoop() {
+            synchronized(this@Arm) {
+                if ( !useVelocityControl) {
+                    when (movementState) {
+                        Arm.MovementState.DOWN -> {
+                            setArmPower(-1.0)
+                            brake.set(DoubleSolenoid.Value.kForward)
+                        }
+                        Arm.MovementState.STATIONARY -> {var targetPos: Double
+                            setArmPower(0.0)
+                            brake.set(DoubleSolenoid.Value.kReverse)
+                        }
+                        Arm.MovementState.UP -> {
+                            setArmPower(1.0)
+                            brake.set(DoubleSolenoid.Value.kForward)
+                        }
+                    }
+                } else {
+                    when (armState) {
+                        Arm.ArmState.LOW -> {
+                            targetPos = -70*Math.PI/360
```

Comment:
you can give enums data, so you could put this `targetPos` into a variable of the arm state